### ISSUE TITLE
Setup failed for raincloud: 1 controller, 2 valves

### ISSUE THREAD

**Issue by siretart** [vanstinator/hass-raincloud#1]
I've installed version 1.0.0-alpha3 using hacs, and kept my configuration from https://www.home-assistant.io/integrations/raincloud/. My setup is one freshly "reworked" melnor aquastat timer, to which I configured two valves. (Adding the 2nd valve broke the built-in Integration, hence my interest in this HACS integration).

This is the output from my homeassistant.log:

```
bash-5.0# grep raincloud home-assistant.log 
2020-05-23 06:14:41 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for raincloud which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant.
2020-05-23 06:16:03 ERROR (SyncWorker_2) [custom_components.raincloud] Unable to connect to Rain Cloud service: 500 Server Error: Internal Server Error for url: https://wifiaquatimer.com/get_cu_and_fu_status?controller_serial=5000000000000faucet_serial=dddd%0A2
2020-05-23 06:16:05 ERROR (MainThread) [homeassistant.setup] Setup failed for raincloud: Integration failed to initialize.
2020-05-23 06:16:05 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform raincloud.switch: Unable to set up component.
2020-05-23 06:16:05 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform raincloud.binary_sensor: Unable to set up component.
2020-05-23 06:16:09 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform raincloud.sensor: Unable to set up component.
```
(controller serial and faucet serial modified for privacy reasons).

It seems to me that the integration is adding a spurios `%0A2` at the end of the request, which the Melnor django app doesn't take kindly (and reports status code 500). Any idea why the URI in the reqeuest may be malformed?

**Comment by vanstinator** [comment]
Thanks for reporting!

This has been a known issue in the `raincloudy` repository for awhile. The good news is it'll be fixed in the next stable release. If you install the current `hass-raincloud` pre-release you'll have the fix _and_ support for both of your valves (which is currently unsupported on the stable release)

**Comment by siretart** [comment]
> If you install the current hass-raincloud pre-release you'll have the fix and support for both of your valves (which is currently unsupported on the stable release)

As mentioned in the original posting, I *am* already using the 1.0.0-alpha3 pre-release, and am getting the same symptom as with the integration provided by homeassistant. How can I make sure what exactl version of raincloud I am running?

Also, I am happy to confirm that I was able to install https://github.com/vanstinator/raincloudy into a virtualenv and query my devices successfully. So the code seems to work fine for me, but may have issues with getting the pre-release on HACS working on my setup. Can you give me some pointers where to look?

**Comment by vanstinator** [comment]
Sorry! My next suggestion would be to uninstall the HACS version. Reboot
HA. Reinstall the pre-release. Reboot HA. I had an instance during
development where HA didn't pull the latest dependency and that dance
worked.

I'm running a single controller and 2 valves here with success on the
pre-release so it _should_ work. If the above doesn't work I'll have to dig
further.

**Comment by vanstinator** [comment]
@siretart looks like the comment I responded to in my email doesn't exist anymore. Did you get yourself sorted?

**Comment by siretart** [comment]
Sorry, I did realize that I accidentally installed 0.0.3 instead of the
alpha version. I managed to correct the issue and now have all 8 zones
controlled in my home assistant. Thank you so much for your work on
raincloudy and hass-raincloud.

The only issue that I noticed are the name of the entities are the ids if
the valves instead of zone names. But I understand that you are already
aware of this. Other than that, seems to work fine. I'll keep testing.
